packages/react-dom/src/__tests__/ReactDOMEventListener-test.js

```diff
@@ -25,62 +25,67 @@ describe('ReactDOMEventListener', () => {
     ReactDOM = require('react-dom');
     // TODO: can we express this test with only public API?
     ReactDOMComponentTree = require('ReactDOMComponentTree');
-    ReactDOMEventListener = require('ReactDOMEventListener');
     ReactTestUtils = require('react-dom/test-utils');
-
-    handleTopLevel = jest.fn();
-    ReactDOMEventListener._handleTopLevel = handleTopLevel;
   });
 
   it('should dispatch events from outside React tree', () => {
-    var otherNode = document.createElement('h1');
-    var component = ReactDOM.render(<div />, document.createElement('div'));
-    expect(handleTopLevel.mock.calls.length).toBe(0);
-    ReactDOMEventListener.dispatchEvent('topMouseOut', {
-      type: 'mouseout',
-      fromElement: otherNode,
-      target: otherNode,
-      srcElement: otherNode,
-      toElement: ReactDOM.findDOMNode(component),
-      relatedTarget: ReactDOM.findDOMNode(component),
-      view: window,
-      path: [otherNode, otherNode],
-    });
-    expect(handleTopLevel.mock.calls.length).toBe(1);
+    var mock = jest.fn();
+
+    var container = document.createElement('div');
+    var stub = ReactDOM.render(<div onMouseOver={mock} />, container);
+    var node = ReactDOM.findDOMNode(stub);
+    var otherComponent = document.createElement('h1');
+    node.appendChild(otherComponent);
+    document.body.appendChild(container);
+
+    var nativeEvent = document.createEvent('Event');
+    nativeEvent.initEvent('mouseover', true, true);
+    otherComponent.dispatchEvent(nativeEvent);
+
+    expect(mock).toBeCalled();
+    document.body.removeChild(container);
```

Comment:
I think the right way to write this test would be something like:

```js
    var mock = jest.fn();

    var container = document.createElement('div');
    var node = ReactDOM.render(<div onMouseEnter={mock} />, container);
    var otherNode = document.createElement('h1');
    document.body.appendChild(container);
    document.body.appendChild(otherNode);

    var nativeEvent = document.createEvent('MouseEvent');
    nativeEvent.initMouseEvent('mouseout', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, node);
    otherNode.dispatchEvent(nativeEvent);
    expect(mock).toBeCalled();
```

This way we verify that if an event originates outside of React tree (on `otherNode`) it still fires an event inside a React tree.

I had to use a different event but that doesn't matter for this test.